test/orange/univ/detail/InfixEvaluator.test.cc

```diff
@@ -0,0 +1,130 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file orange/univ/detail/InfixEvaluator.test.cc
+//---------------------------------------------------------------------------//
+#include "orange/univ/detail/InfixEvaluator.hh"
+
+#include <iomanip>
+
+#include "celeritas_test.hh"
+
+namespace celeritas
+{
+namespace detail
+{
+namespace test
+{
+//---------------------------------------------------------------------------//
+
+using VecSense = std::vector<Sense>;
+
+constexpr auto lbegin = logic::lbegin;
+constexpr auto ltrue = logic::ltrue;
+constexpr auto lor = logic::lor;
+constexpr auto land = logic::land;
+constexpr auto lnot = logic::lnot;
+constexpr auto lpar_open = logic::lpar_open;
+constexpr auto lpar_close = logic::lpar_close;
+constexpr auto lend = logic::lend;
+
+constexpr auto s_in = Sense::inside;
+constexpr auto s_out = Sense::outside;
+
+//---------------------------------------------------------------------------//
+// TESTS
+//---------------------------------------------------------------------------//
+
+TEST(InfixEvaluatorTest, enumeration)
+{
+    EXPECT_GE(ltrue, lbegin);
+    EXPECT_GE(lnot, lbegin);
+    EXPECT_GE(land, lbegin);
+    EXPECT_GE(lor, lbegin);
+    EXPECT_GE(lpar_open, lbegin);
+    EXPECT_GE(lpar_close, lbegin);
+    EXPECT_LT(lbegin, lend);
+
+    EXPECT_EQ('*', to_char(ltrue));
+    EXPECT_EQ('|', to_char(lor));
+    EXPECT_EQ('&', to_char(land));
+    EXPECT_EQ('~', to_char(lnot));
+    EXPECT_EQ('(', to_char(lpar_open));
+    EXPECT_EQ(')', to_char(lpar_close));
+}
+
+TEST(InfixEvaluatorTest, evaluate)
+{
+    // Logic for alpha : !1 | 2 | !3 | 4 | !8
```

Comment:
What about combining the operator with parens? In practice we only have "any of..." or "all of..." and don't mix within a subexpression. This would reduce the string size at the cost of having a local Boolean stack of operators associated with the open parens, I think?

So: `&( 1 2 -3 4 ) |( 5 -6 &(7 8 ) )` where we have "land" and "lor " be combined operator plus open parens

Reply:
That's an option but then this would be the prefix notation, the point of the infix notation is to not require a stack